### PR TITLE
planner: fix prepare plan cache doesn't work for expression index.

### LINKED ISSUE #26697: prepareplan cache doesn't work for expression index
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

Enable the prepare plan cache.
```
create table t(a int, b int, key ((a+b)));
prepare stmt from 'select * from t where a+b = ?';
set @a = 123;
show process_list;
execute stmt using @a;
explain for connection 3;
```
### 2. What did you expect to see? (Required)
IndexRangSscan
### 3. What did you see instead (Required)
TableFullScan
### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: close #26697 

Problem Summary:
We use constItem to check the expression. But it will reject the cache constant. 

### What is changed and how it works?

Remove the constraint `constItem`, I don't think we need it.

### Check List 

Tests 

- [x] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note 

```release-note
None
```

### CODE REVIEW

**Review by xuyifangreeneyes — APPROVED** [review]
lgtm

**Review by Reminiscent — APPROVED** [review]

File: executor/explainfor_test.go [hunk 1]
```diff
@@ -486,3 +487,35 @@ func (s *testPrepareSerialSuite) TestPointGetUserVarPlanCache(c *C) {
...
+	tk.MustExec("create table t(a int, b int, key ((a+b)));")
+	tk.MustExec("prepare stmt from 'select * from t where a+b = ?'")
+	tk.MustExec("set @a = 123")
+	tk.MustExec("execute stmt using @a")
+
+	tkProcess := tk.Se.ShowProcess()
+	ps := []*util.ProcessInfo{tkProcess}
+	tk.Se.SetSessionManager(&mockSessionManager1{PS: ps})
+	c.Assert(tk.MustUseIndex("for connection "+strconv.FormatUint(tkProcess.ID, 10), "expression_index"), IsTrue)
+}
```
Comment (eurekaka) [inline executor/explainfor_test.go:521]: Please complete this test for the reusing of the cached plan, i.e, change the value of `@a`, execute the stmt again, and check the range used in the cached plan is correct.

**Review by eurekaka — APPROVED** [review]

**Pull request merged**